libass/ass_fontconfig.c

```diff
@@ -88,14 +88,25 @@ static void destroy(void *priv)
     free(fc);
 }
 
-static void scan_fonts(FcConfig *config, ASS_FontProvider *provider)
+static void scan_fonts(FcConfig *config, ASS_FontProvider *provider, ASS_Library *lib)
 {
     int i;
     FcFontSet *fonts;
     ASS_FontProviderMetaData meta = {0};
 
-    // get list of fonts
-    fonts = FcConfigGetFonts(config, FcSetSystem);
+    // get list of fonts;
+    // sorting by default pattern prefers regular variants
+    FcPattern *pat = FcPatternCreate();
```

Comment:
Can this fail and return `NULL`?

Reply:
no mention in docs, but yes indeed it can, good catch!
This also affects `cache_fallbacks`